example outputting to Tenere controller

nargetdev opened the pull request:
Still has index out of bounds exceptions with more than 180 points in model.  No problem with less - that's why I changed the grid model to 10x17 . Also had to make `fixtureToIndices( ... )` public, maybe there's a better way, but I feel like fixture should have an indices interface if there isn't one.

mcslee commented on src/heronarts/lx/midi/LXMidiInput.java:
```diff
@@ -118,6 +141,25 @@ public void send(MidiMessage midiMessage, long timeStamp) {
+        tempo.tap(false);
+      }
+      this.midiClockCounter += 1;
+      this.midiClockCounter %= 24;
```
And up top you should:
private final static int MIDI_PULSES_PER_QUARTER_NOTE = 24;
this.midiClockCounter = (this.midiClockCounter + 1) % MIDI_PULSES_PER_QUARTER_NOTE;

mcslee commented on src/heronarts/lx/midi/LXMidiInput.java:
```diff
@@ -118,6 +141,25 @@ public void send(MidiMessage midiMessage, long timeStamp) {
   }
 
+  private void handleClockSignal(ShortMessage sm) {
+    Tempo tempo = this.midiEngine.lx.tempo;
```
This is a style nitpick, but I think it's more readable not to store this in a local variable. Just reference it directly in the switch blocks.

mcslee commented on src/heronarts/lx/midi/LXMidiInput.java:
```diff
@@ -118,6 +141,25 @@ public void send(MidiMessage midiMessage, long timeStamp) {
+      this.midiClockCounter %= 24;
+      break;
+    case ShortMessage.START:
+      tempo.trigger();
```
I'm wondering if you want to also reset this.midiClickCounter to 0 here? There could be cases where midi clock stops being sent, and then a new start() signal is sent without there necessarily ever having been a STOP signal sent.

mcslee commented on src/heronarts/lx/Tempo.java:
```diff
@@ -203,6 +203,25 @@ public void tap() {
+      double beatPeriod = this.lastTap - this.firstTap;
+      setBpm(MINUTE / beatPeriod);
+    } else {
+      tap();
```
This is a confusing construction. I would modify this such that you have:

public void tap() {
  tap(true);
}

Then, in this method, show the logic for both the smoothed version as well as the non-smoothed version.

awhitty commented:
Finally got around to wrapping this up :)

awhitty commented:
Closing this to clean up my `/pulls`

Commit:
example outputting to Tenere controller

Still has index out of bounds exceptions with more than 180 points in model.  No problem with less.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/heronarts/LX?pullRequest=5) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/heronarts/LX?pullRequest=5) before we can accept your contribution.<br/><hr/>**Nathan Argetsinger** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/heronarts/LX?pullRequest=5) it.</sub>

mcslee commented:
Hey dude, slightly confused if this is just a duplicate of https://github.com/heronarts/LX/pull/4 or if there's new stuff in here.

Just FYI - I replied to all your questions via email. Also, if you get a chance, I would really like to know where the OutOfBounds exception is coming from. That def shouldn't be happening, so if you can post a stacktrace of the calls that are causing that, would be great to see.

Thanks man!

nargetdev commented:
Hey Mark just getting to this, It’s been pretty wild times so I haven’t really looked since submitting.  I had submitted #4 before realizing you’d fixed some of the issues so I was gonna fix the issues and PR again.

As for the index out of bounds, it seems to have gone away and I can’t reproduce it.

PR again soon

> On Apr 19, 2018, at 11:54 AM, Mark Slee <notifications@github.com> wrote:
> 
> Hey dude, slightly confused if this is just a duplicate of #4 <https://github.com/heronarts/LX/pull/4> or if there's new stuff in here.
> 
> Just FYI - I replied to all your questions via email. Also, if you get a chance, I would really like to know where the OutOfBounds exception is coming from. That def shouldn't be happening, so if you can post a stacktrace of the calls that are causing that, would be great to see.
> 
> Thanks man!
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/heronarts/LX/pull/5#issuecomment-382843930>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AGSpPL7c7jVgaY6a4Mjy-mj_OsSrospwks5tqN17gaJpZM4TbWcV>.
>
